code/script.js

```diff
@@ -1,47 +1,177 @@
+  userInput.value = ''
+  showFoodOptions();
+  form.removeEventListener('submit' , handleNameInput)
+ }
+
+  form.addEventListener('submit' , handleNameInput) 
+ 
+
+  const showFoodOptions = () => { 
+    showMessage(`What type of yummy cake are you in the mood for today ${names}?`, 'bot')
+    form.innerHTML = `
+    <button id="princessBtn">Princess Cake</button>
+    <button id="pancakeBtn">Pancake Cake</button>
+    <button id="strawberryBtn">Strawberry Cake</button>
+  `
+
+    document
+    .getElementById('princessBtn')
+    .addEventListener('click', () => topping('Princess Cake'))
+    document
+    .getElementById('pancakeBtn')
+    .addEventListener('click', () => topping('Pancake Cake'))
+    document
+    .getElementById('strawberryBtn')
+    .addEventListener('click', () => topping('Strawberry Cake'))
+
+    // setTimeout(() => showCakePieses(), 1000)
+ } 
 
-// Set up your eventlisteners here
+ const topping = (type) => {
+   cakeType = type
+   showMessage(type, 'user')
+   showMessage(`Yummy in my tummy! I also love ${type} What kind of topping would you like?`, 'bot') 
+  
 
-// When website loaded, chatbot asks first question.
-// normally we would invoke a function like this:
-// greeting()
-// But if we want to add a little delay to it, we can wrap it in a setTimeout:
-// setTimeout(functionName, timeToWaitInMilliSeconds)
+    if (type === 'Princess Cake') {
+      form.innerHTML = `
+      <select id="select" class="topping"> 
+      <option value="" selected disabled>Pick your topping!</option>
+      <option value="Icing Sugar">Icing sugar</option>
+      <option value="Strawberries">Strawberries</option>
+      <option value="Marzipan Pigs">Marzipan Pigs</option>
+      </select> 
+      `
+    }
+    else if (type === 'Pancake Cake') {
+      form.innerHTML = `
+        <select id="select" class= topping>
+          <option value="" selected disabled>Pick your topping!</option>
+          <option value="Chocolate Sauce">Chocolate Sauce</option>
+          <option value="Sprinkles">Sprinkles</option>
+          <option value="Whipped Cream">Whipped Cream</option>
+        </select>
+      `
+    } else {
+      form.innerHTML = `
+        <select id="select" class="topping">
+          <option value="" selected disabled>Pick your topping!</option>
+          <option value="Blue Berries">Blue Berries</option>
+          <option value="Chocolate Sprinkles">Chocolate Sprinkles</option>
+          <option value="Banana">Banana</option>
+        </select>
+      `
+    }
+    const select = document.getElementById('select')
+    select.addEventListener('change', () => cakePieces(select.value))
+     
+    // select.value ni skickar med det till nästa funktion men använder det inte där consol logga det.
+  }
+    const cakePieces = (topping) => {
+    cakeTopping = topping
+    showMessage(cakeTopping, 'user')
+    showMessage(`${cakeTopping} is a great choice of topping! How many pieces would you like? ${names}`, 'bot')
```

Comment:
Great that you use the users name again! Feels more personal even if you are talking to a bot :) Maybe put the name before the question mark and a "," after "like" so it displays: How many pieces would you like, Name?